### PR TITLE
Consistent place for accessor property values

### PR DESCRIPTION
**Pull request by Josh-Cena**
### Description

Since we already use the "Syntax" subheading with accessor properties, expanding this convention to every page seems beneficial.

1. It allows us to document any possible side-effects with getter/setters.
2. It allows us to document possible exceptions.
3. It provides a consistent place to add the note that "there's no setter for this property".

### CODE REVIEW

**Comment by Elchi3** [comment]
This is an interesting idea. Would you mind creating a discussion about this first? Right now I think API/ and JavaScript/ docs are aligned because we discussed this 1.5 years ago and came to a resolution: https://github.com/mdn/content/discussions/5992

**Comment by Elchi3** [comment]
Hi @Josh-Cena, I realize you are not re-introducing _syntax blocks_. One of the the options discussed in https://github.com/mdn/content/discussions/5992 was to explicitly spell out getter and setters but then we figured that these would be a lot of words for saying "this is read-only" and especially give beginners are harder time and also makes it all less easy to scan.
I like that you want to make things consistent and move them out of the "Description" section, however I wish it would read easier than talking about getters and setters, and I wish the page outline would be the same as in Web APIs for properties where this section is just called "Value".

I think this PR introduces the following structure for property values:

```
## Syntax
### Return value
The getter for `byteLength` returns an integer whose value is established via the first length parameter of the `ArrayBuffer()` constructor. It returns 0 if this `ArrayBuffer` has been detached.

There is no setter for `byteLength`, so you cannot change this property's value using assignment.
```

I think if it were in API, it would read like this:

```
## Value
The read-only `byteLength` property is an integer whose value is established via the first length parameter of the `ArrayBuffer()` constructor. It returns 0 if this ArrayBuffer has been detached.
```

Hence my question: Should we discuss how we want to consistently do this on MDN? I'd be curios to hear what other think.

**Comment by Josh-Cena** [comment]
IMO, web API and JS docs have different levels of abstraction. The thing about web API is it's theoretically language-agnostic: using the same IDL description, I should be able to implement the DOM spec in Python. Python has different concepts of readonly that doesn't involve getter/setters. Therefore, web API tends to not mention anything that's JS-specific, including the prototype chain or the actual semantics of any property (its writability, configurability, etc.), but only their "surface properties", like inheritance and read-onliness. I support this abstraction.

However, in JS, there are different levels of "read-onliness":

- If something is implemented as an accessor on the prototype, we can use `Object.defineProperty` to define an instance property that's totally different; we just can't use the assignment operator.
- If something is a non-writable configurable data property, we can use `Object.defineProperty` to change its value, but we can't change it to an accessor.
- If something is a non-configurable non-writable data property, then both using `Object.defineProperty` and the assignment operator would fail in all cases.

As a language reference, we often spell out this kind of nuance explicitly. In fact we already expose the concept of "no setter"; it's just currently buried in the description. I advocated for the separation of `javascript-instance-data-property` and `javascript-instance-accessor-property` of page types exactly so I could introduce more tailored page structures in the future.

I would compare this to the use of `{{js_property_attributes}}` table: we don't do this for web API, but we assume in-depth JS users know what "configurable" and "writable" are. It's the same idea as `js_property_attributes` here because getters and setters are property attributes; we are just spelling everything out using prose.

Happy to hear what others think as well.

**Comment by hamishwillee** [comment]
> Happy to hear what others think as well.

FWIW 

While I think the Web API docs are lot easier to read, I don't find these hard to parse. So I am not so concerned about the pattern as long as it is consistent. One benefit is also for writers; the approach forces you to spell out each case, whereas using "read only" as a differentiator can be missed.

Of course it would be better if we could find common ground across web APIs and JavaScript, but I wouldn't be keen to take this pattern into Web APIs, and I trust that @Josh-Cena knows what the audience expects. Certainly we've had less complaining from hard-core JavaScript pedants since he's come on board.

Last of all, I do think this kind of change is worth a discussion first, if only to make sure everyone is aware of the reason, not just those who were lucky enough to see this post.

**Comment by bsmth** [comment]
Hi @Josh-Cena - what can we do to help unblock this work? Shall we close the PR for now and open a discussion? Do you want to get some more feedback on the changes?

**Comment by Josh-Cena** [comment]
I don't think it's necessary to change and the status quo is fine

**Comment by bsmth** [comment]
OK, thank you!
